opencast-backend/annotation-impl/src/main/java/org/opencast/annotation/impl/persistence/ExtendedAnnotationServiceJpaImpl.java

```diff
@@ -1007,7 +1069,15 @@ public ExtendedAnnotationException apply(Exception e) {
    *          value of the ":id" parameter in the named query.
    */
   private <A, B> Option<A> findById(final Function<B, A> toA, final String queryName, final Object id) {
-    return tx(named.<B> findSingle(queryName, id(id))).map(toA);
+    Optional<B> lol = (Optional<B>) tx(em -> {
+      return namedQuery.findOpt(queryName, Pair.of("id", id)).apply(em);
+    });
+    if (lol.isPresent()) {
+      A lel = toA.apply(lol.get());
+      return some(lel);
+    } else {
+      return none();
+    }
   }
```

Comment:
This overload looks like a copypasta of the one below it, which was fine I guess when they both were just one line. Although I would say it was questionable before as well. 🤷‍♀️  One should call through to the other.

However, I think we should just entirely get rid of this overload now and do the function application (i.e. mostly conversion I think) at the call site. That way it's consistent with what you did for `findAllM` vs. `findAllWithBlablabla`.